src/orchard/xref.clj

```diff
@@ -13,21 +32,87 @@
     (symbol? thing) (var-get (find-var thing))
     (fn? thing) thing))
 
-(defn fn-deps
-  "Returns a set with all the functions invoked by `val`.
-  `val` can be a function value, a var or a symbol."
-  {:added "0.5"}
-  [val]
-  (let [val (as-val val)]
-    (set (some->> val class .getDeclaredFields
+(defn- f->sym [^clojure.lang.AFn f]
+  (-> f .getClass .getName repl/demunge symbol))
+
+(defn- fn-source [f]
+  (hunt-down-source (f->sym f)))
+
+(defn- fn-name [^java.lang.Class f]
+  (-> f .getName repl/demunge symbol))
+
+(defonce classbytes (atom {}))
+
+(defn- recompile [ns-sym form]
```

Comment:
Why is "recompilation" needed? This would be a great `;; comment` to add.

Reply:
You are right, I should have explained it in the code.
The idea is to recompile the code and overwrite the classLoader just during that recompilation so that we can get at the bytecode. Sadly there is no other way to do that since the JVM just throws all of that away after loading.
This "should" not interfere with normal execution since the new classes are never used and the proxied loader is thrown away after use.
But hopefully, the recompile solution won't be needed if I can fix the final problem with the second solution.
We might want to keep the code around though since the bytecode can be used for other introspection at a later point.